src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/SharedTableBuilder.java

```diff
@@ -375,9 +379,6 @@ private void setDefaults() {
         if (this.billingMode == null) {
             this.billingMode = BillingMode.PROVISIONED;
         }
-        if (secondaryIndexMapper == null) {
```

Comment:
We were always ignoring what was set before or set here anyway, since build was always setting the index mapper after setDefaults().